src/rgw/rgw_swift_auth.cc

```diff
@@ -524,8 +525,13 @@ void RGW_SWIFT_Auth_Get::execute()
       protocol = "https";
     } else {
       server_port = s->info.env->get("SERVER_PORT");
-      add_port = (strcmp(server_port, "80") != 0);
-      protocol = "http";
+      if (server_port) {
```

Comment:
>   The SERVER_PORT variable MUST be set to the TCP/IP port number on
>    which this request is received from the client.  This value is used
>    in the port part of the Script-URI.
> 
>       SERVER_PORT = server-port
>       server-port = 1*digit
> 
>    Note that this variable MUST be set, even if the port is the default
>    port for the scheme and could otherwise be omitted from a URI.
> 

iiuc, we can just `assert(server_port)` before `add_port = (strcmp(server_port, "80") != 0);`.